go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1936,6 +1958,45 @@ func commonFlows(bridge *bridgeConfiguration) ([]string, error) {
 						fmt.Sprintf("cookie=%s, priority=100, in_port=%s, dl_src=%s, ipv6, "+
 							"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:%s",
 							defaultOpenFlowCookie, netConfig.ofPortPatch, bridgeMacAddress, config.Default.ConntrackZone, netConfig.masqCTMark, ofPortPhys))
+
+					// IP traffic from the OVN network to the host network should be handled normally by the bridge instead of
+					// being forwarded directly to the NIC (prio=100 flow above). This allows pods in the default network to reach
+					// localnet pods on the same node.
+					for _, hostSubnet := range hostSubnets {
+						if hostSubnet.IP.To4() != nil {
+							continue
+						}
+						newFlow := fmt.Sprintf("cookie=%s, priority=102, in_port=%s, dl_src=%s, ipv6, ipv6_dst=%s, "+
+							"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:NORMAL",
+							defaultOpenFlowCookie, netConfig.ofPortPatch, bridgeMacAddress, hostSubnet,
+							config.Default.ConntrackZone, netConfig.masqCTMark)
+
+						if config.Gateway.Mode == config.GatewayModeLocal {
+							newFlow = fmt.Sprintf("cookie=%s, priority=102, in_port=LOCAL, dl_src=%s, ipv6, ipv6_dst=%s, "+
+								"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:NORMAL",
+								defaultOpenFlowCookie, bridgeMacAddress, hostSubnet,
+								config.Default.ConntrackZone, ctMarkHost)
+						}
+
+						dftFlows = append(dftFlows, newFlow)
+					}
+
+					for _, icmpType := range []int{types.NeighborSolicitationICMPType, types.NeighborAdvertisementICMPType} {
+						newFlow := fmt.Sprintf("cookie=%s, priority=102, in_port=%s, dl_src=%s, icmp6, icmpv6_type=%d, "+
+							"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:NORMAL",
+							defaultOpenFlowCookie, netConfig.ofPortPatch, bridgeMacAddress, icmpType,
+							config.Default.ConntrackZone, netConfig.masqCTMark)
+
+						if config.Gateway.Mode == config.GatewayModeLocal {
+							newFlow = fmt.Sprintf("cookie=%s, priority=102, in_port=LOCAL, dl_src=%s, icmp6, icmpv6_type=%d, "+
+								"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:NORMAL",
+								defaultOpenFlowCookie, bridgeMacAddress, icmpType,
+								config.Default.ConntrackZone, ctMarkHost)
+						}
+
+						dftFlows = append(dftFlows, newFlow)
+					}
```

Comment:
Don't we need similar stuff for ipv4 ICMP ? like we ping from default pod network to localnet same node ? 

Also add a comment here similar to the ones befefore and the shared + gateway if statements.

Reply:
This needs more explanation, I forgot to add enough comments in the code, sorry.
So for ICMP (v4) we don't need any special flows: it's on top of IPv4, which we allow explicitly with the `ip` filter in the flow. 

Same goes for ICMPv6, as long as the destination involved is within the host subnet, of course.
However neighbor discovery in IPv6 happens through ICMPv6 messages to a special destination (`ff02::1:ff00:0/104`). The flows I'm adding in this PR match against the host subnet, so all neighbor discovery messages in V6 didn't match against the new flows and were getting redirected to the host NIC, thanks to the existing flow at prio=100.  I discussed this with @trozet and we decided it'd be a better solution to just match against the icmp type and the in-port here.